riscv/src/runtime.rs

```diff
@@ -148,6 +148,48 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
+    pub fn with_keccak(mut self) -> Self {
+        self.add_submachine(
+            "test_data::asm::keccakf::KeccakF",
+            None,
+            "keccakf",
+            [format!(
+                "instr keccakf ~ keccakf.keccakf {};", // operation
+                instr_register_params(0, 25, 25) // 25 registers for the states, covered by 26 registers of riscv
+            )],
+            0,
+            // zero out output registers
+            // TODO: I did this similar to what poseidon did,
+            // but I'm concerned that any preexisting values in the registers aren't saved.
+            // Can I simply leave this blank?
+            (0..25).map(|i| format!("{} <=X= 0;", reg(i))),
```

Comment:
This parameter only exists as a workaround for a current limitation in witgen:
any included machines must be called at least once.

So what this should be doing is calling the `keccakf` instruction and then zeroing out the output registers (as they should be zero when the actual program starts). 
Now, I'm not completely sure how this should work when the input parameters are actually memory addresses (what address should we use? and we probably need to zero out memory after the call).

Or maybe this need of calling machines at least once could be fixed in witgen? Gotta ask @georgwiese

Reply:
Yeah, this get's pretty complicated, unfortunately, because this call would do memory operations that we have to "undo". If we're not using continuations, we could just pass any memory pointers and then zero the resulting region again. But in the presence of continuations it's even weirder, because there are constraints that only paged-in pages can be accessed, so what if there are none (because this chunk doesn't otherwise access memory)?

Probably fixing the requirement for witgen is the best solution. #548 tracks it, but it's also not straight-forward...

Reply:
But why should it call Keccakf first before zeroing out the output registers?

> This parameter only exists as a workaround for a current limitation in witgen: any included machines must be called at least once.
> 
> So what this should be doing is calling the `keccakf` instruction and then zeroing out the output registers (as they should be zero when the actual program starts). Now, I'm not completely sure how this should work when the input parameters are actually memory addresses (what address should we use? and we probably need to zero out memory after the call).
> 
> Or maybe this need of calling machines at least once could be fixed in witgen? Gotta ask @georgwiese

Reply:
Given some updates are needed. I'll leave it as is with a TODO for now.

Reply:
Done.